### ISSUE TITLE
Master doesn't build on OpenBSD (leveldb)

### ISSUE THREAD

**Issue by laanwj** [bitcoin/bitcoin#10670]
It looks like OpenBSD's `eg++` which is based on 4.9.4 (or rather, the binutils based on 2.17) doesn't support the CRC32 instructions introduced with the recent leveldb upgrade:
```
Making all in src
gmake[1]: Entering directory '/home/user/bitcoin/src'
gmake[2]: Entering directory '/home/user/bitcoin/src'
gmake[3]: Entering directory '/home/user/bitcoin'
gmake[3]: Leaving directory '/home/user/bitcoin'
eg++ -std=c++11 -DHAVE_CONFIG_H -I. -I../src/config  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -I./leveldb -DOS_OPENBSD -DLEVELDB_ATOMIC_PRESENT -D__STDC_LIMIT_MACROS  -DLEVELDB_PLATFORM_POSIX -I./leveldb/include -I./leveldb/helpers/memenv -DLEVELDB_PLATFORM_POSIX_SSE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -Wstack-protector -fstack-protector-all  -fPIE -msse4.2 -g -O2 -Wall -Wextra -Wformat -Wvla -Wformat-security -Wno-unused-parameter -MT leveldb/port/leveldb_libleveldb_sse42_a-port_posix_sse.o -MD -MP -MF leveldb/port/.deps/leveldb_libleveldb_sse42_a-port_posix_sse.Tpo -c -o leveldb/port/leveldb_libleveldb_sse42_a-port_posix_sse.o `test -f 'leveldb/port/port_posix_sse.cc' || echo './'`leveldb/port/port_posix_sse.cc
leveldb/port/port_posix_sse.cc: In function 'uint32_t leveldb::port::AcceleratedCRC32C(uint32_t, const char*, size_t)':
leveldb/port/port_posix_sse.cc:59:15: warning: 'ecx' may be used uninitialized in this function [-Wmaybe-uninitialized]
   return (ecx & (1 << 20)) != 0;
               ^
leveldb/port/port_posix_sse.cc:57:26: note: 'ecx' was declared here
   unsigned int eax, ebx, ecx, edx;
                          ^
/tmp//ccBAimlE.s: Assembler messages:
/tmp//ccBAimlE.s:95: Error: no such instruction: `crc32b -1(%rbp),%eax'
/tmp//ccBAimlE.s:121: Error: no such instruction: `crc32q -8(%rbp),%rax'
/tmp//ccBAimlE.s:148: Error: no such instruction: `crc32b -1(%rbp),%eax'
:312: Error: no such instruction: `crc32l 0(%rbp),%eax'
gmake[2]: *** [Makefile:4921: leveldb/port/leveldb_libleveldb_sse42_a-port_posix_sse.o] Error 1
gmake[2]: Leaving directory '/home/user/bitcoin/src'
gmake[1]: *** [Makefile:9224: all-recursive] Error 1
gmake[1]: Leaving directory '/home/user/bitcoin/src'
gmake: *** [Makefile:738: all-recursive] Error 1
```
Likely we need to add some check in the build system whether the instruction can be used?

**Comment by janstary** [comment]
I get the same error on 6.1-current/amd64

**Comment by theuni** [comment]
Taking this one. We just need to do a quick compiler check in configure.

**Comment by janstary** [comment]
Same error on MacOSX 10.6.8 (Intel) as well.

**Comment by janstary** [comment]
On the same machine, version 0.14.2 does not have this problem.

**Comment by laanwj** [comment]
Yes, the 0.14 branch works fine, this was introduced on master

**Comment by laanwj** [comment]
This was fixed by #10806
